### PR TITLE
Feature/keymanager tests

### PR DESCRIPTION
**Pull request by l4l**
### Description of the Change

Test for keys manager

### Benefits

Coverage increase

### Possible Drawbacks 

nothing

### Usage Examples or Tests

`make keys_manager_test && ctest -R keys_manager`

### CODE REVIEW

File: test/module/libs/crypto/keys_manager_test.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+using namespace iroha;
+using namespace boost::filesystem;
+using namespace std::string_literals;
+
+class KeyManager : public ::testing::Test {
+ public:
+  bool create_file(const path &ph, const std::string &contents) {
+    std::ofstream f(ph.c_str(), std::ifstream::binary);
+    if (!f)
```
Comment (x3medima17) [inline test/module/libs/crypto/keys_manager_test.cpp]: Maybe `not` instead of `!`?

File: test/module/libs/crypto/keys_manager_test.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      return false;
+    if (!contents.empty())
```
Comment (x3medima17) [inline test/module/libs/crypto/keys_manager_test.cpp]: Same as above

File: test/module/libs/crypto/keys_manager_test.cpp [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+#define PRIVKEY_SIZE 32
+
+using namespace iroha;
+using namespace boost::filesystem;
+using namespace std::string_literals;
+
+class KeyManager : public ::testing::Test {
+ public:
+  bool create_file(const path &ph, const std::string &contents) {
+    std::ofstream f(ph.c_str(), std::ifstream::binary);
```
Comment (x3medima17) [inline test/module/libs/crypto/keys_manager_test.cpp]: Why `ifstream::binary` instead of `ofstream::binary` ?

Comment (l4l) [reply test/module/libs/crypto/keys_manager_test.cpp]: redundant line actually, thanks for noticing

File: libs/crypto/keys_manager_impl.cpp [hunk 4]
```diff
@@ -52,66 +52,51 @@ namespace iroha {
...
+  nonstd::optional<iroha::keypair_t> KeysManagerImpl::loadKeys() {
+    std::string pub_key;
+    std::string priv_key;
 
-    //TODO 15/11/17 motxx - better way to suppress the evaluation?
-    auto blocking = nonstd::make_optional<iroha::keypair_t>()
-        | loadFile(account_name_ + ".pub", client_pub_key)
-        | loadFile(account_name_ + ".priv", client_priv_key);
+    if (!loadFile(account_name_ + ".pub", pub_key)
+        || !loadFile(account_name_ + ".priv", priv_key))
```
Comment (Warchant) [inline libs/crypto/keys_manager_impl.cpp]: loadFile accepts full path, do not add ".pub" or ".priv" to it

Comment (l4l) [reply libs/crypto/keys_manager_impl.cpp]: Out of this pr, these are only refactorings

Comment (Warchant) [reply libs/crypto/keys_manager_impl.cpp]: Okay, got it

**Pull request closed without merging**